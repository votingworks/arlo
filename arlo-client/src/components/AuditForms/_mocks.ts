import { Audit } from '../../types'

export const mockAudit: Audit = {
  name: 'Primary 2019',
  riskLimit: 10,
  randomSeed: 123123123,

  contests: [
    {
      id: 'contest-1',
      name: 'Contest 1',

      choices: [
        {
          id: 'candidate-1',
          name: 'Candidate 1',
          numVotes: 42,
        },
        {
          id: 'candidate-2',
          name: 'Candidate 2',
          numVotes: 72,
        },
      ],

      totalBallotsCast: 4200,
    },
  ],

  jurisdictions: [
    {
      id: 'adams-county',
      name: 'Adams County',
      contests: ['contest-1'],
      auditBoards: [
        {
          id: 'audit-board-1',
          name: 'Audit Board #1',
          members: [],
        },
        {
          id: 'audit-board-2',
          name: 'Audit Board #2',
          members: [],
        },
      ],
      ballotManifest: {
        filename: 'Adams_County_Manifest.csv',
        numBallots: 123456,
        numBatches: 560,
        uploadedAt: '2019-06-17 11:45:00',
      },
    },
  ],

  rounds: [
    {
      startedAt: '2019-06-17 11:45:00',
      endedAt: '2019-06-17 11:55:00',
      contests: [
        {
          id: 'contest-1',
          endMeasurements: {
            pvalue: 0.085,
            isComplete: false,
          },
          results: {
            'candidate-1': 55,
            'candidate-2': 35,
          },
          sampleSize: 25,
        },
      ],
      jurisdictions: {
        'adams-county': {
          numBallots: 15,
        },
      },
    },
  ],
}

export const statusStates: Audit[] = [
  {
    name: '',
    riskLimit: '',
    randomSeed: '',
    contests: [],
    jurisdictions: [],
    rounds: [],
  },
  {
    contests: [
      {
        choices: [
          {
            id: 'choice-1',
            name: 'choice one',
            numVotes: 792,
          },
          {
            id: 'choice-2',
            name: 'choice two',
            numVotes: 1325,
          },
        ],
        id: 'contest-1',
        name: 'contest name',
        totalBallotsCast: 2123,
        sampleSizeOptions: [
          { size: 269, type: 'ASN' },
          { size: 379, prob: 0.8 },
          { size: 78 },
        ],
      },
    ],
    jurisdictions: [],
    rounds: [],
    name: 'contest name',
    randomSeed: '123456789',
    riskLimit: 1,
  },
  {
    contests: [
      {
        choices: [
          {
            id: 'choice-1',
            name: 'choice one',
            numVotes: 792,
          },
          {
            id: 'choice-2',
            name: 'choice two',
            numVotes: 1325,
          },
        ],
        id: 'contest-1',
        name: 'contest name',
        totalBallotsCast: 2123,
        sampleSizeOptions: [
          { size: 269, type: 'ASN' },
          { size: 379, prob: 0.8 },
          { size: 78 },
        ],
      },
    ],
    jurisdictions: [
      {
        auditBoards: [
          {
            id: 'audit-board-1',
            name: 'Audit Board #1',
            members: [],
          },
        ],
        ballotManifest: {
          filename: null,
          numBallots: null,
          numBatches: null,
          uploadedAt: null,
        },
        contests: ['contest-1'],
        id: 'jurisdiction-1',
        name: 'Jurisdiction 1',
      },
    ],
    rounds: [],
    name: 'contest name',
    randomSeed: '123456789',
    riskLimit: 1,
  },
  {
    contests: [
      {
        choices: [
          {
            id: 'choice-1',
            name: 'choice one',
            numVotes: 792,
          },
          {
            id: 'choice-2',
            name: 'choice two',
            numVotes: 1325,
          },
        ],
        id: 'contest-1',
        name: 'contest name',
        totalBallotsCast: 2123,
        sampleSizeOptions: [
          { size: 269, type: 'ASN' },
          { size: 379, prob: 0.8 },
          { size: 78 },
        ],
      },
    ],
    jurisdictions: [
      {
        auditBoards: [
          {
            id: 'audit-board-1',
            name: 'Audit Board #1',
            members: [],
          },
        ],
        ballotManifest: {
          filename: 'Ballot Manifest May 2019 Election - WYNADOTTE.csv',
          numBallots: 2117,
          numBatches: 10,
          uploadedAt: 'Thu, 18 Jul 2019 16:34:07 GMT',
        },
        contests: ['contest-1'],
        id: 'jurisdiction-1',
        name: 'Jurisdiction 1',
      },
    ],
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: null,
              pvalue: null,
            },
            id: 'contest-1',
            results: {},
            sampleSize: 379,
          },
        ],
        endedAt: null,
        startedAt: 'Thu, 18 Jul 2019 16:34:07 GMT',
      },
    ],
    name: 'contest name',
    randomSeed: '123456789',
    riskLimit: 1,
  },
  {
    contests: [
      {
        choices: [
          {
            id: 'choice-1',
            name: 'choice one',
            numVotes: 792,
          },
          {
            id: 'choice-2',
            name: 'choice two',
            numVotes: 1325,
          },
        ],
        id: 'contest-1',
        name: 'contest name',
        totalBallotsCast: 2123,
        sampleSizeOptions: [
          { size: 269, type: 'ASN' },
          { size: 379, prob: 0.8 },
          { size: 78 },
        ],
      },
    ],
    jurisdictions: [
      {
        auditBoards: [
          {
            id: 'audit-board-1',
            name: 'Audit Board #1',
            members: [],
          },
        ],
        ballotManifest: {
          filename: 'Ballot Manifest May 2019 Election - WYNADOTTE.csv',
          numBallots: 2117,
          numBatches: 10,
          uploadedAt: 'Thu, 18 Jul 2019 16:34:07 GMT',
        },
        contests: ['contest-1'],
        id: 'jurisdiction-1',
        name: 'Jurisdiction 1',
      },
    ],
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: true,
              pvalue: 0.00020431431380638307,
            },
            id: 'contest-1',
            results: {
              'choice-1': 100,
              'choice-2': 167,
            },
            sampleSize: 379,
          },
        ],
        endedAt: 'Thu, 18 Jul 2019 16:59:34 GMT',
        startedAt: 'Thu, 18 Jul 2019 16:34:07 GMT',
      },
    ],
    name: 'contest name',
    randomSeed: '123456789',
    riskLimit: 1,
  },
]

export default mockAudit
