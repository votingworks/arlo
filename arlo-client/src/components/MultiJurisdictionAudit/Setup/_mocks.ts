import { ElementType } from '../../../types'
import { setupStages } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
): {
  prevStage: ISidebarMenuItem
  nextStage: ISidebarMenuItem
  menuItems: ISidebarMenuItem[]
} => {
  const prevTitle = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    title: prevTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextTitle = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    title: nextTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  const menuItems = setupStages.map((s: ElementType<typeof setupStages>) => ({
    title: s,
    active: s === stage,
    activate: jest.fn(),
    state,
  }))
  return { prevStage, nextStage, menuItems }
}

export default relativeStages
