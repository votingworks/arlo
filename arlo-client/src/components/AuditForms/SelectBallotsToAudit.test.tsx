import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import { toast } from 'react-toastify'
import SelectBallotsToAudit from './SelectBallotsToAudit'
import { statusStates, ballotManifest } from './_mocks'
import api from '../utilities'

const apiMock = api as jest.Mock<ReturnType<typeof api>, Parameters<typeof api>>

jest.mock('../utilities')
const toastSpy = jest.spyOn(toast, 'error').mockImplementation()

async function inputAndSubmitForm() {
  const getStatusMock = jest
    .fn()
    .mockImplementationOnce(async () => statusStates[2]) // the POST to /audit/status after jurisdictions
  const updateAuditMock = jest
    .fn()
    .mockImplementationOnce(async () => statusStates[3]) // the POST to /audit/status after manifest

  const { getByTestId, getByLabelText, getByText } = render(
    <SelectBallotsToAudit
      audit={statusStates[1]}
      isLoading={false}
      setIsLoading={jest.fn()}
      updateAudit={updateAuditMock}
      getStatus={getStatusMock}
      electionId="1"
    />
  )

  const manifestInput = getByLabelText('Select manifest...')
  fireEvent.change(manifestInput, { target: { files: [] } })
  fireEvent.blur(manifestInput)
  await wait(() => {
    expect(getByText('You must upload a manifest')).toBeTruthy()
  })
  fireEvent.change(manifestInput, { target: { files: [ballotManifest] } })

  const auditBoardInput: any = getByTestId('audit-boards')
  fireEvent.change(auditBoardInput, { target: { selected: 1 } })

  const sampleSizeInput = getByLabelText(
    '379 samples (80% chance of reaching risk limit and completing the audit in one round)'
  )
  fireEvent.click(sampleSizeInput, { bubbles: true })

  const submitButton = getByText('Select Ballots To Audit')
  fireEvent.click(submitButton, { bubbles: true })

  return [getStatusMock, updateAuditMock]
}

beforeEach(() => {
  apiMock.mockReset()
  toastSpy.mockReset()
})

describe('SelectBallotsToAudit', () => {
  it('renders correctly', () => {
    const { container, rerender } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
        electionId="1"
      />
    )
    expect(container).toMatchSnapshot()

    rerender(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
        electionId="1"
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('handles not having sample size options', () => {
    const statusState = {
      contests: [
        {
          choices: [
            {
              id: 'choice-1',
              name: 'choice one',
              numVotes: '792',
            },
            {
              id: 'choice-2',
              name: 'choice two',
              numVotes: '1325',
            },
          ],
          id: 'contest-1',
          name: 'contest name',
          totalBallotsCast: '2123',
          sampleSizeOptions: [],
        },
      ],
      jurisdictions: [],
      rounds: [],
      name: 'contest name',
      randomSeed: '123456789',
      riskLimit: '1',
    }
    const container = render(
      <SelectBallotsToAudit
        audit={statusState}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
        electionId="1"
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('has radio for selecting sampleSize', () => {
    const { getByText, getByLabelText } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
        electionId="1"
      />
    )

    // all options should be present
    expect(getByText('BRAVO Average Sample Number: 269 samples')).toBeTruthy()
    expect(
      getByText(
        '379 samples (80% chance of reaching risk limit and completing the audit in one round)'
      )
    ).toBeTruthy()
    expect(getByText('78 samples'))

    // correct default should be selected
    expect(
      getByLabelText('BRAVO Average Sample Number: 269 samples').hasAttribute(
        'checked'
      )
    ).toBeTruthy()
  })

  it('changes sampleSize based on audit.rounds.contests.sampleSize', () => {
    const { getByLabelText } = render(
      <SelectBallotsToAudit
        audit={statusStates[4]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
        electionId="1"
      />
    )

    expect(
      getByLabelText(
        '379 samples (80% chance of reaching risk limit and completing the audit in one round)'
      ).hasAttribute('checked')
    ).toBeTruthy()
  })

  it('changes number of audits', () => {
    const { getByTestId } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
        electionId="1"
      />
    )

    const auditBoardInput = getByTestId('audit-boards')
    expect(auditBoardInput).toBeInstanceOf(HTMLSelectElement)
    if (auditBoardInput instanceof HTMLSelectElement) {
      fireEvent.change(auditBoardInput, { target: { selectedIndex: 2 } })
      expect(auditBoardInput.selectedOptions[0].innerHTML).toBe('3')
    }
  })

  it('submits sample size, ballot manifest, and number of audits', async () => {
    apiMock.mockImplementation(async () => ({}))

    const [getStatusMock, updateAuditMock] = await inputAndSubmitForm()

    await wait(() => {
      expect(apiMock).toBeCalledTimes(3)

      expect(apiMock.mock.calls[0][0]).toBe('/audit/sample-size')
      expect(apiMock.mock.calls[0][1]).toMatchObject({
        method: 'POST',
        body: JSON.stringify({
          size: '379',
        }),
        headers: {
          'Content-Type': 'application/json',
        },
      })

      expect(apiMock.mock.calls[1][0]).toBe('/audit/jurisdictions')
      expect(JSON.parse(apiMock.mock.calls[1][1].body as string)).toMatchObject(
        {
          // TODO fix type
          jurisdictions: [
            {
              id: expect.stringMatching(/^[-0-9a-z]+$/),
              name: 'Jurisdiction 1',
              contests: ['contest-1'],
              auditBoards: [
                {
                  id: 'audit-board-1',
                  name: 'Audit Board #1',
                  members: [],
                },
              ],
            },
          ],
        }
      )

      expect(apiMock.mock.calls[2][0]).toBe(
        '/jurisdiction/jurisdiction-1/manifest'
      )

      expect((getStatusMock as jest.Mock).mock.calls.length).toBe(1)
      expect((updateAuditMock as jest.Mock).mock.calls.length).toBe(1)
    })
  })

  it('handles api error on /audit/sample-size', async () => {
    apiMock
      .mockImplementationOnce(() => Promise.reject({ message: 'error' }))
      .mockImplementation(async () => ({}))

    const [getStatusMock, updateAuditMock] = await inputAndSubmitForm()

    await wait(() => {
      expect(apiMock).toBeCalledTimes(1)
      expect(toastSpy).toBeCalledTimes(1)

      expect(getStatusMock).toBeCalledTimes(0)
      expect(updateAuditMock).toBeCalledTimes(0)
    })
  })

  it('handles api error on /audit/jurisdictions', async () => {
    apiMock
      .mockImplementationOnce(async () => ({}))
      .mockImplementationOnce(() => Promise.reject({ message: 'error' }))
      .mockImplementation(async () => ({}))

    const [getStatusMock, updateAuditMock] = await inputAndSubmitForm()

    await wait(() => {
      expect(apiMock).toBeCalledTimes(2)
      expect(toastSpy).toBeCalledTimes(1)

      expect(getStatusMock).toBeCalledTimes(0)
      expect(updateAuditMock).toBeCalledTimes(0)
    })
  })

  it('handles api error on /audit/jurisdiction/:id/manifest', async () => {
    apiMock
      .mockImplementationOnce(async () => ({}))
      .mockImplementationOnce(async () => ({}))
      .mockImplementationOnce(() => Promise.reject({ message: 'error' }))

    const [getStatusMock, updateAuditMock] = await inputAndSubmitForm()

    await wait(() => {
      expect(apiMock).toBeCalledTimes(3)
      expect(toastSpy).toBeCalledTimes(1)

      expect(getStatusMock).toBeCalledTimes(1)
      expect(updateAuditMock).toBeCalledTimes(0)
    })
  })

  it('uses the highest prob value from duplicate sampleSizes', () => {
    statusStates[1].contests[0].sampleSizeOptions = [
      { size: 30, prob: 0.8, type: null },
      { size: 30, prob: 0.9, type: null },
    ]
    const { queryAllByText } = render(
      <SelectBallotsToAudit
        audit={statusStates[1]}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
        electionId="1"
      />
    )

    expect(
      queryAllByText(
        '30 samples (90% chance of reaching risk limit and completing the audit in one round)'
      ).length
    ).toBe(1)
  })

  it('does not display duplicate sampleSize options', () => {
    const statusState = { ...statusStates[1] }
    statusState.contests[0].sampleSizeOptions = [
      { size: 30, prob: null, type: null },
      { size: 30, prob: null, type: null },
    ]
    const { queryAllByText } = render(
      <SelectBallotsToAudit
        audit={statusState}
        isLoading={false}
        setIsLoading={jest.fn()}
        updateAudit={jest.fn()}
        getStatus={jest.fn()}
        electionId="1"
      />
    )

    expect(queryAllByText('30 samples').length).toBe(1)
  })
})
