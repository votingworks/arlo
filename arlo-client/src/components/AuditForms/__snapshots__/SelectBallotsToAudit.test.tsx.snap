// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBallotsToAudit handles not having sample size options 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        data-testid="form-two"
        id="formTwo"
      >
        <hr />
        <div
          class="sc-ifAKCX dUfgxM"
        >
          <h2
            class="bp3-heading sc-bxivhb jFEWwS"
          />
          <h2
            class="bp3-heading sc-bxivhb jFEWwS"
          >
            Select Ballots to Audit
          </h2>
          <div
            class="sc-bdVaJa fdbLH"
          >
            <h3
              class="bp3-heading sc-htpNat cNFUHJ"
            >
              Number of Audit Boards
            </h3>
            <div
              class="sc-bwzfXH izcbHX"
            >
              Set the number of audit boards you with to use.
            </div>
            <div
              class="bp3-html-select"
            >
              <select
                data-testid="audit-boards"
                field="[object Object]"
                form="[object Object]"
                id="auditBoards"
              >
                <option>
                  1
                </option>
                <option>
                  2
                </option>
                <option>
                  3
                </option>
                <option>
                  4
                </option>
                <option>
                  5
                </option>
                <option>
                  6
                </option>
                <option>
                  7
                </option>
                <option>
                  8
                </option>
                <option>
                  9
                </option>
                <option>
                  10
                </option>
                <option>
                  11
                </option>
                <option>
                  12
                </option>
                <option>
                  13
                </option>
                <option>
                  14
                </option>
                <option>
                  15
                </option>
              </select>
              <span
                class="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="sc-bdVaJa fdbLH"
          >
            <h3
              class="bp3-heading sc-htpNat cNFUHJ"
            >
              Ballot Manifest
            </h3>
            <div
              class="sc-bwzfXH izcbHX"
            >
              Click "Browse" to choose the appropriate Ballot Manifest file from your computer
            </div>
            <label
              class="bp3-file-input"
            >
              <input
                accept=".csv"
                name="manifest"
                type="file"
              />
              <span
                class="bp3-file-upload-input"
              >
                Select manifest...
              </span>
            </label>
          </div>
        </div>
        <div
          class="sc-EHOje kPsGPf"
        >
          <button
            class="bp3-button bp3-intent-primary"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Select Ballots To Audit
            </span>
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      data-testid="form-two"
      id="formTwo"
    >
      <hr />
      <div
        class="sc-ifAKCX dUfgxM"
      >
        <h2
          class="bp3-heading sc-bxivhb jFEWwS"
        />
        <h2
          class="bp3-heading sc-bxivhb jFEWwS"
        >
          Select Ballots to Audit
        </h2>
        <div
          class="sc-bdVaJa fdbLH"
        >
          <h3
            class="bp3-heading sc-htpNat cNFUHJ"
          >
            Number of Audit Boards
          </h3>
          <div
            class="sc-bwzfXH izcbHX"
          >
            Set the number of audit boards you with to use.
          </div>
          <div
            class="bp3-html-select"
          >
            <select
              data-testid="audit-boards"
              field="[object Object]"
              form="[object Object]"
              id="auditBoards"
            >
              <option>
                1
              </option>
              <option>
                2
              </option>
              <option>
                3
              </option>
              <option>
                4
              </option>
              <option>
                5
              </option>
              <option>
                6
              </option>
              <option>
                7
              </option>
              <option>
                8
              </option>
              <option>
                9
              </option>
              <option>
                10
              </option>
              <option>
                11
              </option>
              <option>
                12
              </option>
              <option>
                13
              </option>
              <option>
                14
              </option>
              <option>
                15
              </option>
            </select>
            <span
              class="bp3-icon bp3-icon-double-caret-vertical"
              icon="double-caret-vertical"
            >
              <svg
                data-icon="double-caret-vertical"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  double-caret-vertical
                </desc>
                <path
                  d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="sc-bdVaJa fdbLH"
        >
          <h3
            class="bp3-heading sc-htpNat cNFUHJ"
          >
            Ballot Manifest
          </h3>
          <div
            class="sc-bwzfXH izcbHX"
          >
            Click "Browse" to choose the appropriate Ballot Manifest file from your computer
          </div>
          <label
            class="bp3-file-input"
          >
            <input
              accept=".csv"
              name="manifest"
              type="file"
            />
            <span
              class="bp3-file-upload-input"
            >
              Select manifest...
            </span>
          </label>
        </div>
      </div>
      <div
        class="sc-EHOje kPsGPf"
      >
        <button
          class="bp3-button bp3-intent-primary"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Select Ballots To Audit
          </span>
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SelectBallotsToAudit renders correctly 1`] = `
<div>
  <form
    data-testid="form-two"
    id="formTwo"
  >
    <hr />
    <div
      class="sc-ifAKCX dUfgxM"
    >
      <h2
        class="bp3-heading sc-bxivhb jFEWwS"
      />
      <h2
        class="bp3-heading sc-bxivhb jFEWwS"
      >
        Select Ballots to Audit
      </h2>
      <div
        class="sc-bdVaJa fdbLH"
      >
        <h3
          class="bp3-heading sc-htpNat cNFUHJ"
        >
          Estimated Sample Size
        </h3>
        <div
          class="sc-bwzfXH izcbHX"
        >
          Choose the initial sample size for each contest you would like to use for Round 1 of the audit from the options below.
        </div>
        <div
          class="sc-bwzfXH izcbHX"
        >
          <div>
            <label
              class="bp3-control bp3-radio"
            >
              <input
                checked=""
                name="sampleSize[contest-1]"
                type="radio"
                value="269"
              />
              <span
                class="bp3-control-indicator"
              />
              BRAVO Average Sample Number: 
              269 samples
              
            </label>
            <label
              class="bp3-control bp3-radio"
            >
              <input
                name="sampleSize[contest-1]"
                type="radio"
                value="379"
              />
              <span
                class="bp3-control-indicator"
              />
              
              379 samples
               (80% chance of reaching risk limit and completing the audit in one round)
            </label>
            <label
              class="bp3-control bp3-radio"
            >
              <input
                name="sampleSize[contest-1]"
                type="radio"
                value="78"
              />
              <span
                class="bp3-control-indicator"
              />
              
              78 samples
              
            </label>
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa fdbLH"
      >
        <h3
          class="bp3-heading sc-htpNat cNFUHJ"
        >
          Number of Audit Boards
        </h3>
        <div
          class="sc-bwzfXH izcbHX"
        >
          Set the number of audit boards you with to use.
        </div>
        <div
          class="bp3-html-select"
        >
          <select
            data-testid="audit-boards"
            field="[object Object]"
            form="[object Object]"
            id="auditBoards"
          >
            <option>
              1
            </option>
            <option>
              2
            </option>
            <option>
              3
            </option>
            <option>
              4
            </option>
            <option>
              5
            </option>
            <option>
              6
            </option>
            <option>
              7
            </option>
            <option>
              8
            </option>
            <option>
              9
            </option>
            <option>
              10
            </option>
            <option>
              11
            </option>
            <option>
              12
            </option>
            <option>
              13
            </option>
            <option>
              14
            </option>
            <option>
              15
            </option>
          </select>
          <span
            class="bp3-icon bp3-icon-double-caret-vertical"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="sc-bdVaJa fdbLH"
      >
        <h3
          class="bp3-heading sc-htpNat cNFUHJ"
        >
          Ballot Manifest
        </h3>
        <div
          class="sc-bwzfXH izcbHX"
        >
          Click "Browse" to choose the appropriate Ballot Manifest file from your computer
        </div>
        <label
          class="bp3-file-input"
        >
          <input
            accept=".csv"
            name="manifest"
            type="file"
          />
          <span
            class="bp3-file-upload-input"
          >
            Select manifest...
          </span>
        </label>
      </div>
    </div>
    <div
      class="sc-EHOje kPsGPf"
    >
      <button
        class="bp3-button bp3-intent-primary"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Select Ballots To Audit
        </span>
      </button>
    </div>
  </form>
</div>
`;

exports[`SelectBallotsToAudit renders correctly 2`] = `
<div>
  <form
    data-testid="form-two"
    id="formTwo"
  >
    <hr />
    <div
      class="sc-ifAKCX dUfgxM"
    >
      <h2
        class="bp3-heading sc-bxivhb jFEWwS"
      />
      <h2
        class="bp3-heading sc-bxivhb jFEWwS"
      >
        Select Ballots to Audit
      </h2>
      <div
        class="sc-bdVaJa fdbLH"
      >
        <h3
          class="bp3-heading sc-htpNat cNFUHJ"
        >
          Estimated Sample Size
        </h3>
        <div
          class="sc-bwzfXH izcbHX"
        >
          Choose the initial sample size for each contest you would like to use for Round 1 of the audit from the options below.
        </div>
        <div
          class="sc-bwzfXH izcbHX"
        >
          <div>
            <label
              class="bp3-control bp3-radio"
            >
              <input
                checked=""
                name="sampleSize[contest-1]"
                type="radio"
                value="269"
              />
              <span
                class="bp3-control-indicator"
              />
              BRAVO Average Sample Number: 
              269 samples
              
            </label>
            <label
              class="bp3-control bp3-radio"
            >
              <input
                name="sampleSize[contest-1]"
                type="radio"
                value="379"
              />
              <span
                class="bp3-control-indicator"
              />
              
              379 samples
               (80% chance of reaching risk limit and completing the audit in one round)
            </label>
            <label
              class="bp3-control bp3-radio"
            >
              <input
                name="sampleSize[contest-1]"
                type="radio"
                value="78"
              />
              <span
                class="bp3-control-indicator"
              />
              
              78 samples
              
            </label>
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa fdbLH"
      >
        <h3
          class="bp3-heading sc-htpNat cNFUHJ"
        >
          Number of Audit Boards
        </h3>
        <div
          class="sc-bwzfXH izcbHX"
        >
          Set the number of audit boards you with to use.
        </div>
        <div
          class="bp3-html-select"
        >
          <select
            data-testid="audit-boards"
            field="[object Object]"
            form="[object Object]"
            id="auditBoards"
          >
            <option>
              1
            </option>
            <option>
              2
            </option>
            <option>
              3
            </option>
            <option>
              4
            </option>
            <option>
              5
            </option>
            <option>
              6
            </option>
            <option>
              7
            </option>
            <option>
              8
            </option>
            <option>
              9
            </option>
            <option>
              10
            </option>
            <option>
              11
            </option>
            <option>
              12
            </option>
            <option>
              13
            </option>
            <option>
              14
            </option>
            <option>
              15
            </option>
          </select>
          <span
            class="bp3-icon bp3-icon-double-caret-vertical"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="sc-bdVaJa fdbLH"
      >
        <h3
          class="bp3-heading sc-htpNat cNFUHJ"
        >
          Ballot Manifest
        </h3>
        <div
          class="sc-bwzfXH izcbHX"
        >
          Click "Browse" to choose the appropriate Ballot Manifest file from your computer
        </div>
        <label
          class="bp3-file-input"
        >
          <input
            accept=".csv"
            name="manifest"
            type="file"
          />
          <span
            class="bp3-file-upload-input"
          >
            Select manifest...
          </span>
        </label>
      </div>
    </div>
    <div
      class="bp3-spinner"
    >
      <div
        class="bp3-spinner-animation"
      >
        <svg
          height="50"
          stroke-width="8.00"
          viewBox="1.00 1.00 98.00 98.00"
          width="50"
        >
          <path
            class="bp3-spinner-track"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
          />
          <path
            class="bp3-spinner-head"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
            pathLength="280"
            stroke-dasharray="280 280"
            stroke-dashoffset="210"
          />
        </svg>
      </div>
    </div>
  </form>
</div>
`;
